pkg/asset/agent/image/oc.go

```diff
@@ -288,6 +293,40 @@ func (r *release) verifyCacheFile(image, file, architecture string) (bool, error
 	return false, nil
 }
 
+// Remove any existing files in the cache.
+func removeCacheFile(path string) error {
+	matches, err := filepath.Glob(path)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range matches {
+		if err = os.Remove(file); err != nil {
+			return err
+		}
+		logrus.Debugf("Removed file %s", file)
+	}
+	return nil
+}
+
+func getExtractedFileFromWildcard(path string) (string, error) {
+	matches, err := filepath.Glob(path)
+	if err != nil {
+		return "", err
+	}
+
+	for _, file := range matches {
```

Comment:
I think in this case that having more than one match could be potentially a problem, so at least a warning could be useful under such condition

Reply:
Sure, added a check for more than one files that aren't symlinks